core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -74,6 +81,10 @@ class TaskInfo(
     gettingResultTime = time
   }
 
+  private[spark] def markKilledAttempt {
```

Comment:
It is better to change the method signature to `def markKilledByOtherAttempt(): Unit = {`, since this method has side affect, it is better to add parenthesis.